source/core/src/main/com/csse3200/game/components/player/InventoryComponent.java

```diff
@@ -1,183 +1,383 @@
 package com.csse3200.game.components.player;
 
-import java.awt.Point; // for positional data
-import java.util.ArrayList;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
-
+import com.badlogic.gdx.graphics.Texture;
+import com.csse3200.game.components.Component;
+import com.csse3200.game.components.items.ItemComponent;
+import com.csse3200.game.entities.Entity;
+import com.csse3200.game.entities.EntityType;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.badlogic.gdx.utils.Json;
-import com.csse3200.game.components.Component;
-import com.csse3200.game.entities.Entity;
+import java.util.HashMap;
+import java.util.List;
 
 /**
  * A component intended to be used by the player to track their inventory.
  * Currently untested, but forms the basis for the UI which will be implemented soon:tm:
  */
 public class InventoryComponent extends Component {
+  private static final Logger logger = LoggerFactory.getLogger(InventoryComponent.class);
+  //private List<Entity> inventory = new ArrayList<>();
+  private HashMap<String, Integer> itemCount = new HashMap<>();
 
-  public int findFirstIndex() {
-    for (int i = 0; i < itemPlace.size();i++) {
-      if (itemPlace.getOrDefault(i,null) == null) {
-        return i;
-      }
-      else {
-        if (i == 30) {
-          return -1;
-        }
-      }
+  private HashMap<String,Entity> heldItemsEntity = new HashMap<>();
```

Comment:
Could all 4 of these hashmaps not be combined into 1? I don't get why we need 4 different maps where we can have one map containing all the data.